stats/data_access/riot.py

```diff
@@ -42,4 +42,17 @@ def get_matches(self, puuid):
         return self.request_url(self.get_matches_url(puuid) + self.api_key_url + self.get_matches_path_url())
 
     def get_match(self, match_id):
-        return self.request_url(self.get_match_url() + match_id + self.api_key_url)
\ No newline at end of file
+        try:
+            return self.request_url(self.get_match_url() + match_id + self.api_key_url)
```

Comment:
This functions sleeps the interpreter processing if you are under your rate limit time.  This may not be what you want in a high concurrency environment, or even as a service attached to a UI.

I think there are better ways to handle resiliency in python.  Was looking for something similar to Polly in .NET, this article kind of has something: https://medium.com/@glizzykingdreko/an-introduction-to-resilient-caller-simplify-your-python-code-2edd2b086bb9 

Even still, if this is to be used as a web server to support many callers, you need an async pipeline.  I don't immediately see it, is django doing this for you automatically?

Reply:
Put this in ToDos, will do some more work on this, i think Django Rest Framework might help